tcl/org.testeditor.tcl.model/src/org/testeditor/tcl/util/TclModelUtil.xtend

```diff
@@ -44,4 +52,22 @@ class TclModelUtil extends TmlModelUtil {
 		]
 	}
 
+	def dispatch Iterable<TestStep> getTestSteps(ComponentTestStepContext context) {
+		return context.steps
+	}
+
+	def dispatch Iterable<TestStep> getTestSteps(MacroTestStepContext context) {
+		return #[context.step]
+	}
+
+	def Set<EnvParam> getEnvParams(EObject object) {
+		val root = EcoreUtil.getRootContainer(object)
+		if (root instanceof TclModel) {
+			if (root.requiredEnvParams !== null) {
+				return root.requiredEnvParams.envParams.toSet
```

Comment:
There is no need to convert them to a set here. Maybe the called wants the order in which they were defined.